proxy/http/HttpTransact.cc

```diff
@@ -7834,7 +7834,13 @@ HttpTransact::build_response(State *s, HTTPHdr *base_response, HTTPHdr *outgoing
                              HTTPStatus status_code, const char *reason_phrase)
 {
   if (reason_phrase == nullptr) {
-    reason_phrase = http_hdr_reason_lookup(status_code);
+    if (status_code != HTTP_STATUS_NONE) {
+      reason_phrase = http_hdr_reason_lookup(status_code);
+      Debug("http_transact", "Using reason phrase from status %d: %s", status_code, reason_phrase);
+    } else if (base_response->status_get() != HTTP_STATUS_NONE) {
```

Comment:
We may want to check the availability of `base_response`, although it doesn't make sense if it has `nullptr` on this line.

Reply:
I agree. We're best off making sure it isn't nullptr to be safe.